world/expedition.cpp

```diff
@@ -0,0 +1,606 @@
+			expedition_members.character_id
+		FROM expedition_details
+			INNER JOIN instance_list ON expedition_details.instance_id = instance_list.id
+			INNER JOIN expedition_members ON expedition_members.expedition_id = expedition_details.id
+		ORDER BY expedition_details.id;
+	);
+
+	auto results = database.QueryDatabase(query);
+	if (results.Success())
+	{
+		uint32_t last_expedition_id = 0;
+
+		for (auto row = results.begin(); row != results.end(); ++row)
+		{
+			uint32_t expedition_id = strtoul(row[0], nullptr, 10);
+
+			if (last_expedition_id != expedition_id)
+			{
+				expeditions.emplace_back(
+					static_cast<uint32_t>(strtoul(row[0], nullptr, 10)), // expedition_id
+					static_cast<uint32_t>(strtoul(row[1], nullptr, 10)), // dz_instance_id
+					static_cast<uint32_t>(strtoul(row[2], nullptr, 10)), // dz_zone_id
+					static_cast<uint32_t>(strtoul(row[3], nullptr, 10)), // start_time
+					static_cast<uint32_t>(strtoul(row[4], nullptr, 10))  // duration
+				);
+			}
+
+			last_expedition_id = expedition_id;
+
+			uint32_t member_id = static_cast<uint32_t>(strtoul(row[5], nullptr, 10));
+			expeditions.back().AddMember(member_id);
+		}
+	}
+
+	return expeditions;
+}
+
+Expedition ExpeditionDatabase::LoadExpedition(uint32_t expedition_id)
+{
+	LogExpeditions("Loading expedition [{}] for world cache", expedition_id);
+
+	Expedition expedition;
+
+	std::string query = fmt::format(SQL(
+		SELECT
+			expedition_details.id,
+			expedition_details.instance_id,
+			instance_list.zone,
+			instance_list.start_time,
+			instance_list.duration,
+			expedition_members.character_id
+		FROM expedition_details
+			INNER JOIN instance_list ON expedition_details.instance_id = instance_list.id
+			INNER JOIN expedition_members ON expedition_members.expedition_id = expedition_details.id
+		WHERE expedition_details.id = {};
+	), expedition_id);
+
+	auto results = database.QueryDatabase(query);
+	if (results.Success())
+	{
+		bool created = false;
+		for (auto row = results.begin(); row != results.end(); ++row)
+		{
+			if (!created)
+			{
+				expedition = Expedition{
+					static_cast<uint32_t>(strtoul(row[0], nullptr, 10)), // expedition_id
+					static_cast<uint32_t>(strtoul(row[1], nullptr, 10)), // dz_instance_id
+					static_cast<uint32_t>(strtoul(row[2], nullptr, 10)), // dz_zone_id
+					static_cast<uint32_t>(strtoul(row[3], nullptr, 10)), // start_time
+					static_cast<uint32_t>(strtoul(row[4], nullptr, 10))  // duration
+				};
+				created = true;
+			}
+
+			auto member_id = static_cast<uint32_t>(strtoul(row[5], nullptr, 10));
+			expedition.AddMember(member_id);
+		}
+	}
+
+	return expedition;
+}
+
+void ExpeditionDatabase::DeleteExpeditions(const std::vector<uint32_t>& expedition_ids)
+{
+	LogExpeditionsDetail("Deleting [{}] expedition(s)", expedition_ids.size());
+
+	std::string expedition_ids_query;
+	for (const auto& expedition_id : expedition_ids)
+	{
+		fmt::format_to(std::back_inserter(expedition_ids_query), "{},", expedition_id);
```

Comment:
How do you feel about just using `fmt::join` for this instead since it's templated to work on ranges of any formattable type
<https://fmt.dev/latest/api.html#_CPPv2N3fmt4joinERK5Range11string_view>